src/Rendering/prePassRenderer.ts

```diff
@@ -265,17 +265,18 @@ export class PrePassRenderer {
      * @param subMesh Submesh on which the effect is applied
      */
     public bindAttachmentsForEffect(effect: Effect, subMesh: SubMesh) {
+        const material = subMesh.getMaterial();
+        const isPrePassCapable = material && material.isPrePassCapable;
+        const excluded = material && this.excludedMaterials.indexOf(material) !== -1;
+
         if (this.enabled && this._currentTarget.enabled) {
-            if (effect._multiTarget) {
+            if (effect._multiTarget && isPrePassCapable && !excluded) {
                 this._engine.bindAttachments(this._multiRenderAttachments);
             } else {
                 this._engine.bindAttachments(this._defaultAttachments);
 
-                if (this._geometryBuffer && this.currentRTisSceneRT) {
-                    const material = subMesh.getMaterial();
-                    if (material && !material.isPrePassCapable && this.excludedMaterials.indexOf(material) === -1) {
-                        this._geometryBuffer.renderList!.push(subMesh.getRenderingMesh());
-                    }
+                if (this._geometryBuffer && this.currentRTisSceneRT && !excluded) {
+                    this._geometryBuffer.renderList!.push(subMesh.getRenderingMesh());

```

Comment:
That will add the mesh into `renderList` each frame (in case the `if` condition is `true`)? `bindAttachmentsForEffect` is called by the `scene._beforeRenderingMeshStage` observable.

Reply:
yes indeed, but we clear the renderlist before each render pass here https://github.com/BabylonJS/Babylon.js/blob/b540dcdcff44fe30ced78241bb4a2f0d77ed7ba7/src/Rendering/prePassRenderer.ts#L389

Reply:
Is there a way to prevent GC ? maybe using renderList.length = 0 instead ???